glommio/src/io/dma_file.rs

```diff
@@ -163,11 +163,7 @@ impl DmaFile {
             | (opts.custom_flags as libc::c_int & !libc::O_ACCMODE);
 
         let res = DmaFile::open_at(dir, path, flags, opts.mode).await;
-        let mut f = enhanced_try!(res, opdesc, Some(path), None)?;
-        // FIXME: Don't assume 512 or 4096, we can read this info from sysfs
-        // currently, we just use the minimal {values which make sense}
-        f.o_direct_alignment = if opts.write { 4096 } else { 512 };
```

Comment:
@glommer this PR removes this in favor of always using the alignment from sysfs. It will be `512` in most cases. I don't remember why we used `4096` when the write bit is set. Was it for compatibility?

Reply:
512 writes are expensive even if allowed. The logical block is not 512-bytes, so you have to RMW internally.

Reply:
Also, my memory is failing me a bit here, but I don't think the OS will like a sub-4096 write even if the media allows it.

I'd like to be defensive here and keep writes at a minimum of 4k.

Reply:
Ok. One thing that bothers me is that we have one alignment setting per file so if a file is opened with the write bit, then reads will be aligned on 4k too. We could change that though.

As for write alignment, there are two write patterns:
* Most people use linear writes and use large buffers, much larger than the minimum. Does it make a difference if a 128k buffer is aligned on 512 or 4k? When you talk about alignment I feel you compound that with the minimum IO buffer size. Can we make 4k buffers the minimum with a 512b alignment if that's what sysfs says? 
* A few people will use random writes and for them, it makes sense to expose the very minimum alignment the system supports IMO. 

What do you think?

Reply:
> 512 writes are expensive even if allowed. The logical block is not 512-bytes, so you have to RMW internally.

Even if sysfs puts `512` in `logical_block_size`? That sounds wild

Reply:
I don't think the random writes case is very interesting at all. It seldom shows up.